tests/suite/reconfig_test.go

```diff
@@ -0,0 +1,638 @@
+		err = writeReconfigResults(outFile, results)
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	When("resources exist before startup", func() {
+		testDescription := "Test 1: Resources exist before startup"
+
+		It("gathers metrics after creating 30 resources", func() {
+			resourceCount := 30
+			timeToReadyStartingLogSubstring := "Starting NGINX Gateway Fabric"
+			test := createResourcesGWLast(resourceCount)
+
+			Expect(test).To(Succeed())
+			Expect(checkResourceCreation(resourceCount)).To(Succeed())
+
+			ngfPodName, startTime := deployNGFReturnsNGFPodNameAndStartTime()
+
+			collectMetrics(
+				testDescription,
+				resourceCount,
+				timeToReadyStartingLogSubstring,
+				ngfPodName,
+				startTime,
+			)
+		})
+
+		It("gathers metrics after creating 150 resources", func() {
+			resourceCount := 150
+			timeToReadyStartingLogSubstring := "Starting NGINX Gateway Fabric"
+			test := createResourcesGWLast(resourceCount)
+
+			Expect(test).To(Succeed())
+			Expect(checkResourceCreation(resourceCount)).To(Succeed())
+
+			ngfPodName, startTime := deployNGFReturnsNGFPodNameAndStartTime()
+
+			collectMetrics(
+				testDescription,
+				resourceCount,
+				timeToReadyStartingLogSubstring,
+				ngfPodName,
+				startTime,
+			)
+		})
+	})
+
+	When("NGF and Gateway resource are deployed first", func() {
+		testDescription := "Test 2: Start NGF, deploy Gateway, create many resources attached to GW"
+
+		It("gathers metrics after creating 30 resources", func() {
+			resourceCount := 30
+			timeToReadyStartingLogSubstring := "Reconciling the resource\",\"controller\":\"httproute\""
+			test := createResourcesRoutesLast(resourceCount)
+
+			ngfPodName, startTime := deployNGFReturnsNGFPodNameAndStartTime()
+
+			Expect(test).To(Succeed())
+			Expect(checkResourceCreation(resourceCount)).To(Succeed())
+
+			collectMetrics(
+				testDescription,
+				resourceCount,
+				timeToReadyStartingLogSubstring,
+				ngfPodName,
+				startTime,
+			)
+		})
+
+		It("gathers metrics after creating 150 resources", func() {
+			resourceCount := 150
+			timeToReadyStartingLogSubstring := "Reconciling the resource\",\"controller\":\"httproute\""
+			test := createResourcesRoutesLast(resourceCount)
+
+			ngfPodName, startTime := deployNGFReturnsNGFPodNameAndStartTime()
+
+			Expect(test).To(Succeed())
+			Expect(checkResourceCreation(resourceCount)).To(Succeed())
+
+			collectMetrics(
+				testDescription,
+				resourceCount,
+				timeToReadyStartingLogSubstring,
+				ngfPodName,
+				startTime,
+			)
+		})
+	})
+
+	When("NGF and resources are deployed first", func() {
+		testDescription := "Test 3: Start NGF, create many resources attached to a Gateway, deploy the Gateway"
+
+		It("gathers metrics after creating 30 resources", func() {
+			resourceCount := 30
+			timeToReadyStartingLogSubstring := "Reconciling the resource\",\"controller\":\"gateway\""
+			test := createResourcesGWLast(resourceCount)
+
+			ngfPodName, startTime := deployNGFReturnsNGFPodNameAndStartTime()
```

Comment:
same question, won't the gateway still exist before NGF does?

Reply:
Same reasoning as above